src/main/java/sirius/biz/storage/layer1/replication/jdbc/SQLReplicationTaskStorage.java

```diff
@@ -0,0 +1,211 @@
+
+    @ConfigValue("storage.layer1.replication.maxBatches")
+    private int maxBatches;
+
+    @ConfigValue("storage.layer1.replication.replicateDeleteDelay")
+    private Duration replicateDeleteDelay;
+
+    @ConfigValue("storage.layer1.replication.replicateUpdateDelay")
+    private Duration replicateUpdateDelay;
+
+    @ConfigValue("storage.layer1.replication.retryReplicationDelay")
+    private Duration retryReplicationDelay;
+
+    @ConfigValue("storage.layer1.replication.maxReplicationAttempts")
+    private int maxReplicationAttempts;
+
+    @Part
+    private SQLEntityBatchEmitter emitter;
+
+    @Part
+    private OMA oma;
+
+    @Part
+    private Mixing mixing;
+
+    @Part
+    private ReplicationManager replicationManager;
+
+    @Part
+    private DistributedTasks distributedTasks;
+
+    @Override
+    public void notifyAboutDelete(String primarySpace, String objectId) {
+        oma.select(SQLReplicationTask.class)
+           .eq(SQLReplicationTask.PRIMARY_SPACE, primarySpace)
+           .eq(SQLReplicationTask.OBJECT_KEY, objectId)
+           .delete();
+
+        SQLReplicationTask task = new SQLReplicationTask();
+        task.setPrimarySpace(primarySpace);
+        task.setObjectKey(objectId);
+        task.setPerformDelete(true);
+        task.setEarliestExecution(LocalDateTime.now().plus(replicateDeleteDelay));
+        oma.update(task);
+    }
+
+    @Override
+    public void notifyAboutUpdate(String primarySpace, String objectId) {
+        oma.select(SQLReplicationTask.class)
+           .eq(SQLReplicationTask.PRIMARY_SPACE, primarySpace)
+           .eq(SQLReplicationTask.OBJECT_KEY, objectId)
+           .delete();
+
+        SQLReplicationTask task = new SQLReplicationTask();
+        task.setPrimarySpace(primarySpace);
+        task.setObjectKey(objectId);
+        task.setPerformDelete(false);
+        task.setEarliestExecution(LocalDateTime.now().plus(replicateUpdateDelay));
+        oma.update(task);
+    }
+
+    @Override
+    public int emitBatches() {
+        Database database = oma.getDatabase(mixing.getDescriptor(SQLReplicationTask.class).getRealm());
+        AtomicInteger numberOfBatches = new AtomicInteger(maxBatches);
+        AtomicInteger scheduledTasks = new AtomicInteger(0);
+        emitter.computeBatches(SQLReplicationTask.class, qry -> enhanceQuery(qry, true), batchSize, batch -> {
+            distributedTasks.submitFIFOTask(ReplicationTaskExecutor.class, batch);
+            scheduledTasks.addAndGet(batchSize);
+
+            markTasksAsScheduled(database, batch);
+
+            return numberOfBatches.decrementAndGet() > 0;
+        });
+
+        return scheduledTasks.get();
+    }
+
+    private void markTasksAsScheduled(Database database, JSONObject batch) {
+        try {
+            database.createQuery("UPDATE sqlreplicationtask "
+                                 + "SET scheduled = ${scheduled} "
+                                 + "WHERE id >= ${lowerBound}"
```

Comment:
formatting inside the query (spaces on line end mixed with spaces on line start)